homeassistant/components/coolmaster/climate.py

```diff
@@ -54,6 +61,9 @@ async def async_setup_entry(
 
     async_add_devices(all_devices)
 
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(RESET_FILTER, {}, "async_reset_filter")
```

Comment:
I think the current best practice and modern way to achieve something like this is not to add more to the existing climate entity, but instead add additional entities. For example:
- A sensor or binary sensor entity to show the state of the filter needing to be cleaned
- A button entity to provide a reset
- Additional sensor for the error code state

I just had a look at other integrations doing this and found `sensibo` doing this.

My impression is the motivation behind this is to make it easier for end users to compose these things with other parts of home assistant. For example, the UI for helping with automations. Or by having these entities as `config` or `diagnostic` entity types they show up as options in the UI associated with the underlying device.

Reply:
I see you left a separate comment about this.  I don't think i agree with the rationale as good reasons to go against the current design standards. 

My impression is that the naming issue you describe is handled by giving the device a descriptive names and the current approach is to name entities based on the device name:
https://developers.home-assistant.io/blog/2022/07/10/entity_naming?_highlight=entity&_highlight=name